### ISSUE TITLE
[Bug]: StorageState Not Used VSCode Debug Mode

### ISSUE THREAD

**Issue by st11x** [microsoft/playwright#31620]
### Version

1.45.0

### Steps to reproduce

1. Create an auth.setup.ts file as per https://playwright.dev/docs/auth#basic-shared-account-in-all-tests. (Login details changed as per my app)
2. Write test that assumes login state. E.g.:

```js
test("Go to Landing", async ({ page }) => {
    await page.goto("/ARM/");
    await expect(page.getByRole("button", { name: "User Functions" })).toBeVisible();
});

```

3. In VSCode, debug the test.

### Expected behavior

The test should pass as the page context is logged in.

### Actual behavior

The test failed. The browser shows the login page.

### Additional context

I can see that the authenticate setup function is called each time, then the browser is closed, then the actual test launches but the login context is gone. The user.json file is created and is populated with the necessary cookies.

If I just ran the test in VSCode **without** the Debug mode, it works fine. Again, the authenticate step is called before the test and everything works as intended.

Finally, I ran the authenticate method only in normal mode. Without closing the browser (call it A), filter out the setup from the test as per https://playwright.dev/docs/auth#authenticating-in-ui-mode, then run the actual test in Debug mode, it works fine.

Debug mode will actually close browser A first, and it will still work. However, if I close any of the Debug mode browsers, it stops working again. 

### Environment

```shell
System:
    OS: Windows 10 10.0.19045
    CPU: (16) x64 11th Gen Intel(R) Core(TM) i7-11850H @ 2.50GHz
    Memory: 14.77 GB / 31.73 GB
  Binaries:
    Node: 22.3.0 - C:\bin\node\node.EXE
    npm: 10.8.1 - C:\bin\node\npm.CMD
  Languages:
    Bash: 4.4.12 - C:\Users\masn\AppData\Local\Temp\Mxt241\bin\bash.EXE
  npmPackages:
    @playwright/test: ^1.45.0 => 1.45.0
```

**Comment by yury-s** [comment]
Can you share complete reproduction example?

**Comment by st11x** [comment]
Other than the code given, it is hard to share a reproduction as the system is non-public.

Let me see if I can create one using a public system, but essentially, it is just a simple login page as in the authenticate documentation, even down to how it is wired up. 

The test fails immediately on the first line because it doesn't have a session when the debugger is used.

**Comment by yury-s** [comment]
Where do you store the storage state file?

**Comment by st11x** [comment]
I created a different example that authenticates to this practice site at https://practice.expandtesting.com/login.
Debug mode on this one does work properly so it looks like some incompatibilities with the internal site.

There is nothing much to go on, so I'll close this ticket and open another when I can find the cause.

**Comment by st11x** [comment]
> Where do you store the storage state file?
> 
> I tried with this simple project and it works fine: https://github.com/yury-s/bug-31620

I followed the docs and set the file to .auth/user.json. The test works in headless mode, in Run mode (VSCode), just not VSCode Debug mode.  Debug mode is definitely reading the file because if I emptied it out (blank file), the test will not start and will complain about the file. 

Thanks for setting up that minimal example.

**Comment by st11x** [comment]
I have made some further observations regarding this particular app. In the first line of the test script, I had to navigate to the dashboard. 

```js
    await page.goto("/ARM/");
    ...
```

This causes the app to lose the session. Before I started converting the authentication from test.beforeEach, I didn't have to navigate to the dashboard because the beforeEach left the browser open. I found that the app will lose the session if you also refresh the browser. 

Now with the test structured like that, running the test in VSCode with both the setup and test projects selected seems to work (test passes).  

![image](https://github.com/microsoft/playwright/assets/7398499/2fad9ee2-c087-4b69-87df-356d142f968e)

The test will always run through the set up, close the browser, then start the test. It'll still end up losing the session at the end.

In Debug mode, it will always fail. 

In headless CLI mode, it passes. 

In headed CLI mode, it fails.

In debug CLI mode, it fails. 

Don't know if there are many apps are like that but it is probably an edge case.
